### PR TITLE
[FLOSS H2] New keyCopy example

### PR DESCRIPTION
**Pull request by muskater**
I have added a new example program to illustrate the usage of `keyCopy`.

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by markus2330** [comment]
jenkins build libelektra please

File: examples/keyCopyExample.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+#include <kdb.h>
+#include <stdio.h>
+#include <string.h>
+
+
+void printNameAndContent (Key * k, char * varName)
+{
+	char buffer[256];
+	keyGetString (k, buffer, sizeof (buffer));
```
Comment (markus2330) [inline examples/keyCopyExample.c:17]: keyGetString is deprecated and shouldn't be used in examples, see #3975

Any specific reason why you preferred keyGetString over keyString?

Comment (muskater) [reply examples/keyCopyExample.c:17]: I was not aware that it is deprecated. Thank you for pointing this out.

**Comment by markus2330** [comment]
Please rebase, there is a conflict in a file. Please set "ready to merge" if you are ready to merge.

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
The build server is not yet happy:

```
 In file included from /Applications/Xcode_13.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h:194,
                 from /Users/runner/work/libelektra/libelektra/examples/keyCopyExample.c:11:
/Users/runner/work/libelektra/libelektra/examples/keyCopyExample.c: In function 'printNameAndContent':
/Users/runner/work/libelektra/libelektra/examples/keyCopyExample.c:17:9: error: '__builtin_strncpy' specified bound 256 equals destination size [-Werror=stringop-truncation]
   17 |         strncpy (buffer, keyString (k), sizeof (buffer));
      |         ^~~~~~~
cc1: all warnings being treated as errors
```

**Comment by muskater** [comment]
I have fixed the error and rebased the PR

**Comment by markus2330** [comment]
Thank you for the example! :sparkle:

**Pull request merged**